### DISCUSSION TITLE
Dead Packages?

### DISCUSSION THREAD

**Discussion by shellscape** [denoland/deno#11809] [Q&A] [answered]
Hey was just wondering if there's a procedure for packages which have been published to deno.land but are now defunct. For example: https://deno.land/x/dex@1.0.2. The repository linked to for this package no longer exists.

**Comment by lucacasonato** [comment, marked as answer]
No. They stay around forever 🙂

**Reply by shellscape** [reply]
that's a bummer, especially so when early adopters create zombie packages that take up precious real estate, like dex. npm suffers from the same.

**Reply by lucacasonato** [reply]
The upside is that you get a immutable registry you can rely on us not interfering with.

**Reply by shellscape** [reply]
I wouldn't necessarily call that an upside, but definitely a side effect that could be relied upon. still, it seems that there should be some kind of a flag that denotes that the repo no longer backs the package in the registry. I'm still new to the ecosystem, but everything I've read about it being security-centric would suggest that there is a lack of confidence in the package registry if there's no guarantee the backing repository or project still exists.

**Reply by lucacasonato** [reply]
The goal of the /x registry is to provide a registry that the ecosystem can rely on. We do this by immutably hosting modules that are uploaded via GitHub. The point is that if the GitHub repo goes away, the package is **not** removed from the registry (uploaded files are immutable). You can rely on modules on /x still being around in a few years time, even if the author has lost interest in their project.

**Reply by shellscape** [reply]
I see. That explanation makes more sense, thanks for elaborating. (would be great content for the docs, btw!)

I'd also love to see some guidance from deno around continuing the work of an abandoned package. Naming, licensing, etc.

**Reply by MierenManz** [reply]
it would be nice for maintainers to put in when a project has died and that a new maintainer would be needed for it to keep existing.

**Reply by max-hk** [reply]
How about NPM takes on releasing unused package name?

They keep old version immuned, but allow new maintainers to take over the package name and release newer semver versions of the package. (If I recalled it correctly)
